bundles/binding/org.openhab.binding.sunnywebbox/src/main/java/org/openhab/binding/sunnywebbox/internal/SunnyWebBoxGenericBindingProvider.java

```diff
@@ -0,0 +1,94 @@
+/**
+ * Copyright (c) 2010-2015, openHAB.org and others.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+ 
+package org.openhab.binding.sunnywebbox.internal;
+
+import org.openhab.binding.sunnywebbox.SunnyWebBoxBindingProvider;
+import org.openhab.core.binding.BindingConfig;
+import org.openhab.core.items.Item;
+import org.openhab.model.item.binding.AbstractGenericBindingProvider;
+import org.openhab.model.item.binding.BindingConfigParseException;
+
+
+/**
+ * This class is responsible for parsing the binding configuration.
```

Comment:
briefly explain what this class is about, also give some examples of the binding configuration here